app/src/main/java/org/simple/clinic/drugs/search/DrugSearchListItem.kt

```diff
@@ -0,0 +1,89 @@
+package org.simple.clinic.drugs.search
+
+import androidx.paging.PagingData
+import androidx.paging.TerminalSeparatorType.SOURCE_COMPLETE
+import androidx.paging.insertSeparators
+import androidx.paging.map
+import androidx.recyclerview.widget.DiffUtil
+import io.reactivex.subjects.Subject
+import org.simple.clinic.R
+import org.simple.clinic.databinding.ListItemDrugSearchBinding
+import org.simple.clinic.widgets.PagingItemAdapter
+import org.simple.clinic.widgets.recyclerview.BindingViewHolder
+
+sealed class DrugSearchListItem : PagingItemAdapter.Item<DrugSearchListItem.DrugClicked> {
+
+  companion object {
+
+    fun from(searchResults: PagingData<Drug>): PagingData<DrugSearchListItem> {
+      return searchResults
+          .map(::DrugSearchResult)
+          .insertSeparators(SOURCE_COMPLETE) { oldItem, newItem ->
+            if (oldItem != null && newItem != null) {
+              Divider
+            } else {
+              null
+            }
+          }
+    }
+  }
+
+  data class DrugClicked(val drug: Drug)
+
+  data class DrugSearchResult(val drug: Drug) : DrugSearchListItem() {
+
+    override fun layoutResId() = R.layout.list_item_drug_search
+
+    override fun render(holder: BindingViewHolder, subject: Subject<DrugClicked>) {
+      val binding = holder.binding as ListItemDrugSearchBinding
+
+      binding.drugItemCard.setOnClickListener { subject.onNext(DrugClicked(drug)) }
+
+      binding.drugNameTextView.text = drugName(drug)
+    }
+
+    private fun drugName(drug: Drug): String {
+      return buildString {
+        append(drug.name)
+
+        if (!drug.dosage.isNullOrBlank()) {
+          append(", ${drug.dosage}")
+        }
+
+        if (drug.frequency != null) {
+          append(", ${drug.frequency}")
+        }
```

Comment:
Can we clarify if appending commas to display strings is applicable to all the languages that we support?

Reply:
Yeah, I thought about moving the "," into strings for translation. But currently the languages we support don't have a different comma notation. So hard coded it, but we can move it to strings just to be safe.

Reply:
It's not about whether they have a comma notation or not, it's about whether the notion of a comma exists in those languages as well. I feel this is something that should be left up to the localizers to decide.